VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -68,7 +86,78 @@ public void deleteProfileImage(@NonNull String username) throws RetroHttpExcepti
         return userService.getBadges(username, page, ApiConstants.STANDARD_PAGE_SIZE);
     }
 
-    public List<EnrolledCoursesResponse> getUserEnrolledCourses(@NonNull String username) throws RetroHttpException {
-        return userService.getUserEnrolledCourses(username);
+    public List<EnrolledCoursesResponse> getUserEnrolledCourses(@NonNull String username, boolean tryCache) throws RetroHttpException {
+        List<EnrolledCoursesResponse> ret = null;
+        String json = null;
+
+        final String cacheKey = config.getApiHostURL() + "/api/mobile/v0.5/users/" + username + "/course_enrollments";
+
+        // try to get from cache if we should
+        if (tryCache) {
+            try {
+                json = cache.get(cacheKey);
+            }
+            catch (IOException | NoSuchAlgorithmException e) {
+                logger.debug(e.toString());
+            }
+        }
+
+        // if we don't have a json yet, get it from userService
+        if (json == null) {
+            Response response = null;
+            try {
+                response = userService.getUserEnrolledCourses(username);
+            }
+            catch (Throwable e) {
+                logger.debug(e.toString());
+            }
+
+            if (response != null) {
+                TypedInput input = response.getBody();
+                if (input != null) {
+                    try {
+                        InputStream is = input.in();
+                        json = fromStream(is);
+                        is.close();
+                    }
+                    catch (IOException e) {
+                        json = null;
+                        logger.debug(e.toString());
+                    }
+                }
+            }
+        }
+
+        if (json != null) {
+
+            // cache result
+            try {
+                cache.put(cacheKey, json);
+            }
+            catch (IOException | NoSuchAlgorithmException e) {
+                logger.debug(e.toString());
+            }
+
+            // NOTE: trying to pass in List<EnrolledCourseResponse> (with the help of token type) yields NoClassDefFoundError
+            ret = new ArrayList<>();
+            JsonArray ary = gson.fromJson(json, JsonArray.class);
+            for (int cnt=0; cnt<ary.size(); ++cnt) {
+                ret.add(gson.fromJson(ary.get(cnt), EnrolledCoursesResponse.class));
+            }
+        }
+
+        return ret;
+    }
+
+    private static String fromStream(InputStream in) throws IOException {
```

Comment:
Ah, left that in by accident! Thanks.

On Thu, May 26, 2016 at 3:04 PM, Brian Guertin <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java
> <https://github.com/edx/edx-app-android/pull/707#discussion_r64802196>:
>
> > +            catch (IOException | NoSuchAlgorithmException e) {
> > +                logger.debug(e.toString());
> > +            }
> > +
> > +            // NOTE: trying to pass in List<EnrolledCourseResponse> (with the help of token type) yields NoClassDefFoundError
> > +            ret = new ArrayList<>();
> > +            JsonArray ary = gson.fromJson(json, JsonArray.class);
> > +            for (int cnt=0; cnt<ary.size(); ++cnt) {
> > +                ret.add(gson.fromJson(ary.get(cnt), EnrolledCoursesResponse.class));
> > +            }
> > +        }
> > +
> > +        return ret;
> > +    }
> > +
> > +    private static String fromStream(InputStream in) throws IOException {
>
> this method can be replaced with IOUtils.toString(in)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/707/files/dbeabd58cd79988fd33dab0d0c53394a097b3c57#r64802196>
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX